src/Raven.Server/Web/System/AdminTcpConnectionDebugInfoHandler.cs

```diff
@@ -0,0 +1,112 @@
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Extensions;
+
+namespace Raven.Server.Web.System
+{
+    public class AdminTcpConnectionDebugInfoHandler : RequestHandler
+    {
+        [RavenAction("/admin/debug/info/tcp/stats", "GET", AuthorizationStatus.Operator, IsDebugInformationEndpoint = true)]
+        public Task Statistics()
+        {
+            var properties = TcpExtensions.GetIPGlobalPropertiesSafely();
+            var ipv4Stats = properties.GetTcpIPv4StatisticsSafely();
+            var ipv6Stats = properties.GetTcpIPv6StatisticsSafely();
+
+            var djv = new DynamicJsonValue
+            {
+                ["IPv4"] = ToDynamic(ipv4Stats),
+                ["IPv6"] = ToDynamic(ipv6Stats)
+            };
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(writer, djv);
+            }
+
+            return Task.CompletedTask;
+
+            static DynamicJsonValue ToDynamic(TcpStatistics stats)
+            {
+                if (stats == null)
+                    return null;
+
+                var result = new DynamicJsonValue();
+                result[nameof(stats.ConnectionsAccepted)] = stats.GetConnectionsAcceptedSafely();
+                result[nameof(stats.ConnectionsInitiated)] = stats.GetConnectionsInitiatedSafely();
+                result[nameof(stats.CumulativeConnections)] = stats.GetCumulativeConnectionsSafely();
+                result[nameof(stats.CurrentConnections)] = stats.GetCurrentConnectionsSafely();
+                result[nameof(stats.ErrorsReceived)] = stats.GetErrorsReceivedSafely();
+                result[nameof(stats.FailedConnectionAttempts)] = stats.GetFailedConnectionAttemptsSafely();
+                result[nameof(stats.MaximumConnections)] = stats.GetMaximumConnectionsSafely();
+                result[nameof(stats.MaximumTransmissionTimeout)] = stats.GetMaximumTransmissionTimeoutSafely();
+                result[nameof(stats.MinimumTransmissionTimeout)] = stats.GetMinimumTransmissionTimeoutSafely();
+                result[nameof(stats.ResetConnections)] = stats.GetResetConnectionsSafely();
+                result[nameof(stats.ResetsSent)] = stats.GetResetsSentSafely();
+                result[nameof(stats.SegmentsReceived)] = stats.GetSegmentsReceivedSafely();
+                result[nameof(stats.SegmentsResent)] = stats.GetSegmentsResentSafely();
+                result[nameof(stats.SegmentsSent)] = stats.GetSegmentsSentSafely();
+
+                return result;
+            }
+        }
+
+        [RavenAction("/admin/debug/info/tcp/active-connections", "GET", AuthorizationStatus.Operator, IsDebugInformationEndpoint = true)]
+        public Task ActiveConnections()
+        {
+            var properties = TcpExtensions.GetIPGlobalPropertiesSafely();
```

Comment:
This can return null

Reply:
Every extension method that I've added checks that.